DatadogSessionReplay/Tests/Mocks/TestScheduler.swift

```diff
@@ -19,6 +19,8 @@ internal class TestScheduler: Scheduler {
     /// Queue to execute operations on.
     let queue: Queue
 
+    var isRunning = false
```

Comment:
Because `start()` / `stop()` and `isRunning` (getter) are called from different threads in tests, we're running into data race that will be causing flaky tests in long run:

<img width="1334" alt="Screenshot 2023-06-20 at 09 59 52" src="https://github.com/DataDog/dd-sdk-ios/assets/2358722/0809c0bb-91dd-4641-95eb-b606bb71c735">